Algorithm.Framework/Portfolio/EqualWeightingPortfolioConstructionModel.py

```diff
@@ -29,15 +29,22 @@ class EqualWeightingPortfolioConstructionModel(PortfolioConstructionModel):
     For insights of direction InsightDirection.Up, long targets are returned and
     for insights of direction InsightDirection.Down, short targets are returned.'''
 
-    def __init__(self, resolution = Resolution.Daily):
+    def __init__(self, rebalancingParam = Resolution.Daily):
         '''Initialize a new instance of EqualWeightingPortfolioConstructionModel
         Args:
-            resolution: Rebalancing frequency'''
+            rebalancingParam: Rebalancing parameter. It can be a function, timedelta or Resolution'''
         self.insightCollection = InsightCollection()
         self.removedSymbols = []
         self.nextExpiryTime = UTCMIN
         self.rebalancingTime = UTCMIN
-        self.rebalancingPeriod = Extensions.ToTimeSpan(resolution)
+        self.rebalancingFunc = rebalancingParam
+
+        # If the argument is an instance if Resolution or Timedelta
+        # Redefine self.rebalancingFunc
+        if isinstance(rebalancingParam, int):
+            self.rebalancingFunc = lambda dt: dt + Extensions.ToTimeSpan(rebalancingParam)
```

Comment:
Maybe we could move `Extensions.ToTimeSpan(rebalancingParam)` outside of the lambda so we don't run it each time